cli/azd/cmd/infra_create.go

```diff
@@ -82,10 +113,7 @@ func (ica *infraCreateAction) Run(
 		return err
 	}
 
-	formatter := output.GetFormatter(ctx)
-	writer := output.GetWriter(ctx)
-
-	infraManager, err := provisioning.NewManager(ctx, env, prj.Path, prj.Infra, !ica.rootOptions.NoPrompt)
+	infraManager, err := provisioning.NewManager(ctx, env, prj.Path, prj.Infra, !i.flags.global.NoPrompt)
```

Comment:
I would love to see us be able to inject the `Environment`, `Project` & `provisioning.Manager` without having to do this dance in each action.

I have a simplified example @ 
- https://github.com/wbreza/wire-sample/blob/main/src/pkg/actions/provision.go
- https://github.com/wbreza/wire-sample/blob/main/src/pkg/actions/wire.go

Reply:
Excellent observation and suggestion. That is coming -- I'm trying to minimize these changes to purely the refactor without changing behavior too much, so each component will be tackled separately in multiple PRs.

Reply:
Great - do we have issues in the backlog created to address this? If not can you please add?

Reply:
#899